src/queries/subjects.ts

```diff
@@ -1,13 +1,35 @@
+import { useState, useEffect } from "react";
+
 export interface Subject {
   name: string;
   lastDate: string;
 }
 
 export function useSubjects(): Subject[] | null {
-  const subjects = [
-    { name: "PBL演習", lastDate: "2023-01-01" },
-    { name: "サイバーセキュリティ", lastDate: "2023-01-02" },
-    // その他の科目...
-  ];
+  const [subjects, setSubjects] = useState<Subject[] | null>(null);
+
+  useEffect(() => {
+    const controller = new AbortController();
+    const signal = controller.signal;
+
+    const SUBJECTS_ENDPOINT = "https://backend.mikuroxina.workers.dev/subjects";
```

Comment:
```suggestion
    const SUBJECTS_ENDPOINT = "https://backend.mikuroxina.workers.dev/me/subjects";
```
このままだとうまく動かない & フロントエンドだけで直せる方法も無かったので, API として `/me/subjects` を追加しました. こちらをお使いください.